File: java-base/src/main/java/net/cryptic_game/backend/base/utils/ValidationUtils.java
package net.cryptic_game.backend.base.utils;

import java.util.regex.Pattern;

public final class ValidationUtils {

    private ValidationUtils() {
        throw new UnsupportedOperationException();
    }

    public static boolean checkPassword(final String password) {
        return Pattern.compile("(?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).{8,}").matcher(password).find();
    }
}

File: java-base/src/main/kotlin/net/cryptic_game/backend/base/utils/ValidationUtils.kt
package net.cryptic_game.backend.base.utils

import org.apache.commons.validator.routines.EmailValidator

object ValidationUtils {

    private val pattern = "(?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).{8,}".toRegex()

    @JvmStatic
    fun checkMail(mail: String): Boolean = EmailValidator.getInstance().isValid(mail)

    @JvmStatic
    fun checkPassword(password: String) = pattern.matches(password)
}